### PR TITLE
feat: Add Array Support to EXTR serializer

### PR DESCRIPTION
**Pull request by arnelplim**
# Required for all PRs

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #

### CODE REVIEW

**Comment by arnelplim** [comment]
Initial review.  Still need to add test cases to extr_test.go

**Comment by arnelplim** [comment]
> Looks good to me, but I can't read GO :) Will test this on the VOSS LLDP implementation, and confirm it works as expected

I'll hold off merging this back in into extr_1.21 until I get confirmation from you.

File: plugins/serializers/extr/README.md [hunk 1]
```diff
@@ -9,26 +9,29 @@ The `extr` output data format converts metrics into JSON documents, performing t
...
         --> "tags":{partNumber:"1647G-00129 800751-00-01", revision:"01"}
+        *You could also use partNumber_tags="1647G-00129 800751-00-01", revision_tags="01"
 
 - Groups like metric names into a toplevel map. Name of group is same as name, but with first char lowercase
         "fanStats" :[{grouped_FanStats_Metric1}, {grouped_FanStats_Metric2} ]
 
 - Creates nested JSON schema by parsing underscore "_" seperated field keys
-        cpu1_subcore_core_keys=2
-        cpu2_subcore_core_keys=5
+        cpu1_subcore_core_key=2
```
Comment (vullanatt) [inline plugins/serializers/extr/README.md:28]: Are we planning to change the CPU utilization message? Wondering what this change is - "keys to key"

**Review by vullanatt — APPROVED** [review]

File: plugins/serializers/extr/README.md (hunk 1 above)
Comment (arnelplim) [reply plugins/serializers/extr/README.md:28]: No changes.  Right now any fields ending with _key (i.e. x1_key, x2_key)  is grouped into {keys: { x1, x2 }} grouping.  Same goes for any field with _tag which generates {tags:{x1, x2}}.  Thus, _key and _tag suffix are "special" suffixes in that key becomes keys and tag becomes tags.

That said, when I added support for nested JSON you could "also" just end field names with _keys, which would also create the same JSON format. (i.e. keys : { ...}).

**Review by dshnayder — APPROVED** [review]

**Comment by apopescu7** [comment]
This was tested with LLDP events and looks good.
Thanks for adding this.

**Pull request merged**